bitbots_vision/src/bitbots_vision/vision_modules/field_boundary.py

```diff
@@ -629,13 +657,59 @@ def _calculate_field_boundary(_image, _field_color_detector, _x_steps, _y_steps,
         return _field_boundary_points
 
 
+class CVReversedFieldBoundaryAlgorithm(FieldBoundaryAlgorithm):
+    @staticmethod
+    def _calculate_field_boundary(image, field_color_detector, x_steps, y_steps, roi_height, roi_width, roi_increase, green_threshold):
+        # type: () -> list
+        """
+        Finds the points of the field boundary visible in the image. Uses the reversed method iterating from bottom to
+        top until it finds enough non green points. Useful for when two fields are adjacent to each other.
+
+        :returns: list of field boundary points
+        """
+        # calculate the field_mask which contains 0 for non-green pixels and 255 for green pixels in the image
+        # index counting up from top to bottom and left to right
+        field_mask = field_color_detector.get_mask_image()
+
+        # Scale the image down
+        subsampled_mask = cv2.resize(field_mask,(x_steps,y_steps), interpolation=cv2.INTER_AREA)
+        # Define the blur kernel
+        kernel = (2 * roi_height + 1, 2 * roi_width + 1)
+        # Blur the downscaled image to fill holes in the field mask
+        subsampled_mask = cv2.GaussianBlur(subsampled_mask, kernel, 0)
+
+        field_boundary_points = []
+
+        # Iterate horizontal over the image
+        for x_position in range(subsampled_mask.shape[1]):
+            # Iterate vertical over the downscaled mask
+            for y_position in range(subsampled_mask.shape[0]):
+                # Invert the current vertical value
+                max_y = (subsampled_mask.shape[0] - 1) - y_position
+                # Check if we found a pixel under the threshold
+                if subsampled_mask[max_y, x_position] < int(green_threshold / 1000 * 255):
+                    # Reset to last step
+                    max_y += 1
+                    break
+
+            # Scale the stuff for the original image + black magic
+            field_boundary_points.append(
+                (int((x_position + 0.5) * (field_mask.shape[1] / x_steps)),
+                 int(max_y * (field_mask.shape[0] / y_steps))))
+        # Fix tuff at the image edges
+        field_boundary_points[0] = (0, field_boundary_points[0][1])
+        field_boundary_points[-1] = (field_mask.shape[1]-1, field_boundary_points[0][1])
```

Comment:
Are you sure that you want `field_boundary_points[0][1] ` here instead of `field_boundary_points[-1][1]` ?

Reply:
Oh yeah I fix it